### ISSUE TITLE
GDScript: crash when incomplete inner class used

### ISSUE THREAD

**Issue by ThakeeNathees** [godotengine/godot#40856]
**Godot version:** v4.0.dev.custom_build 5ba46ddf8

**OS/device including version:** windows 10

**Issue description:**
the following code crash the editor (copy and paste it in to a script and save it)
```gdscript
class A:
	var b:B = B.new()

class B:
```
backtrace
```
CrashHandlerException: Program crashed
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] HashMap<StringName,int,HashMapHasherDefault,HashMapComparatorDefault<StringName>,3,8>::get (C:\dev\__test_env\godot\core\hash_map.h:283)
[1] HashMap<StringName,int,HashMapHasherDefault,HashMapComparatorDefault<StringName>,3,8>::operator[] (C:\dev\__test_env\godot\core\hash_map.h:424)
[2] GDScriptParser::ClassNode::get_member (C:\dev\__test_env\godot\modules\gdscript\gdscript_parser.h:565)
[3] GDScriptAnalyzer::reduce_identifier_from_base (C:\dev\__test_env\godot\modules\gdscript\gdscript_analyzer.cpp:2064)
[4] GDScriptAnalyzer::reduce_identifier (C:\dev\__test_env\godot\modules\gdscript\gdscript_analyzer.cpp:2152)
[5] GDScriptAnalyzer::reduce_array (C:\dev\__test_env\godot\modules\gdscript\gdscript_analyzer.cpp:1284)
[6] GDScriptAnalyzer::reduce_call (C:\dev\__test_env\godot\modules\gdscript\gdscript_analyzer.cpp:1796)
[7] GDScriptAnalyzer::reduce_array (C:\dev\__test_env\godot\modules\gdscript\gdscript_analyzer.cpp:1272)
[8] GDScriptAnalyzer::resolve_class_interface (C:\dev\__test_env\godot\modules\gdscript\gdscript_analyzer.cpp:488)
[9] GDScriptAnalyzer::resolve_class_interface (C:\dev\__test_env\godot\modules\gdscript\gdscript_analyzer.cpp:622)
[10] GDScriptAnalyzer::resolve_program (C:\dev\__test_env\godot\modules\gdscript\gdscript_analyzer.cpp:3098)
[11] GDScriptAnalyzer::analyze (C:\dev\__test_env\godot\modules\gdscript\gdscript_analyzer.cpp:3116)
[12] GDScriptLanguage::lookup_code (C:\dev\__test_env\godot\modules\gdscript\gdscript_editor.cpp:2945)
[13] ScriptTextEditor::_validate_symbol (C:\dev\__test_env\godot\editor\plugins\script_text_editor.cpp:864)
[14] call_with_variant_args_helper<ScriptTextEditor,String const &,0> (C:\dev\__test_env\godot\core\callable_method_pointer.h:132)
[15] call_with_variant_args<ScriptTextEditor,String const &> (C:\dev\__test_env\godot\core\callable_method_pointer.h:158)
[16] CallableCustomMethodPointer<ScriptTextEditor,String const &>::call (C:\dev\__test_env\godot\core\callable_method_pointer.h:185)
[17] Callable::call (C:\dev\__test_env\godot\core\callable.cpp:50)
[18] Object::emit_signal (C:\dev\__test_env\godot\core\object.cpp:1170)
[19] Object::emit_signal (C:\dev\__test_env\godot\core\object.cpp:1225)
[20] TextEdit::_gui_input (C:\dev\__test_env\godot\scene\gui\text_edit.cpp:2429)
[21] MethodBind1<TextEdit,Ref<InputEvent> const &>::call (C:\dev\__test_env\godot\core\method_bind.gen.inc:867)
[22] Object::call_multilevel (C:\dev\__test_env\godot\core\object.cpp:737)
[23] Object::call_multilevel (C:\dev\__test_env\godot\core\object.cpp:836)
[24] Viewport::_gui_input_event (C:\dev\__test_env\godot\scene\main\viewport.cpp:2309)
[25] Viewport::input (C:\dev\__test_env\godot\scene\main\viewport.cpp:2969)
[26] Window::_window_input (C:\dev\__test_env\godot\scene\main\window.cpp:911)
[27] call_with_variant_args_helper<Window,Ref<InputEvent> const &,0> (C:\dev\__test_env\godot\core\callable_method_pointer.h:132)
[28] call_with_variant_args<Window,Ref<InputEvent> const &> (C:\dev\__test_env\godot\core\callable_method_pointer.h:158)
[29] CallableCustomMethodPointer<Window,Ref<InputEvent> const &>::call (C:\dev\__test_env\godot\core\callable_method_pointer.h:185)
[30] Callable::call (C:\dev\__test_env\godot\core\callable.cpp:50)
[31] DisplayServerWindows::_dispatch_input_event (C:\dev\__test_env\godot\platform\windows\display_server_windows.cpp:1752)
[32] DisplayServerWindows::_dispatch_input_events (C:\dev\__test_env\godot\platform\windows\display_server_windows.cpp:1729)
[33] Input::_parse_input_event_impl (C:\dev\__test_env\godot\core\input\input.cpp:614)
[34] Input::parse_input_event (C:\dev\__test_env\godot\core\input\input.cpp:440)
[35] Input::flush_accumulated_events (C:\dev\__test_env\godot\core\input\input.cpp:830)
[36] DisplayServerWindows::process_events (C:\dev\__test_env\godot\platform\windows\display_server_windows.cpp:1491)
[37] OS_Windows::run (C:\dev\__test_env\godot\platform\windows\os_windows.cpp:627)
[38] widechar_main (C:\dev\__test_env\godot\platform\windows\godot_windows.cpp:163)
[39] _main (C:\dev\__test_env\godot\platform\windows\godot_windows.cpp:185)
[40] main (C:\dev\__test_env\godot\platform\windows\godot_windows.cpp:199)
[41] __scrt_common_main_seh (d:\A01\_work\6\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[42] BaseThreadInitThunk
-- END OF BACKTRACE --

```

**Comment by lyuma** [comment]
I ran into this yesterday. Can you check again now that pull request #40824 was merged?

**Comment by ThakeeNathees** [comment]
@lyuma  yeah, it was fixed after  #40824 merged
